x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -507,12 +518,26 @@ public boolean isActive() {
     /**
      * Checks whether the given feature is allowed, tracking the last usage time.
      */
+    @SuppressForbidden(reason = "Argument to Math.abs() is definitely not Long.MIN_VALUE")
     public boolean checkFeature(Feature feature) {
         boolean allowed = isAllowed(feature);
         LongAccumulator maxEpochAccumulator = lastUsed.get(feature);
+        long now = System.currentTimeMillis();
+        long licenseExpirationDate = getLicenseExpiryDate();
         if (maxEpochAccumulator != null) {
             maxEpochAccumulator.accumulate(epochMillisProvider.getAsLong());
         }
+
+        if (feature.minimumOperationMode.compareTo(OperationMode.BASIC) > 0 &&
+            now >  licenseExpirationDate - LICENSE_EXPIRATION_WARNING_PERIOD.getMillis()) {
+            final long days = TimeUnit.MILLISECONDS.toDays(licenseExpirationDate - now);
+            final String expiryMessage = days == 0? "expires today":
+                (days > 0? String.format(Locale.ROOT, "will expire in [%d] days", days):
+                    String.format(Locale.ROOT, "has expired [%d] days ago", Math.abs(days)));
```

Comment:
For expired licenses, I think it's preferable to just print out the date that is expired, rather than calculate how lonh ago that was.
That's what the log warning does and it seems neater for this case.

> LICENSE [EXPIRED] ON [SUNDAY, NOVEMBER 29, 2020]

The number of days makes sense for "will expire" because it creates a sense of urgency, but once it's expired I think the date is a better option.

Reply:
Changed the wording to "license expired on [EEEE, MMMM DD, YYYY]". For the sace of keeping the headers relatively short kept the rest of the message the same.